core/remotes/docker/httpreadseeker.go

```diff
@@ -76,7 +77,7 @@ func (hrs *httpReadSeeker) Read(p []byte) (n int, err error) {
 		if _, err2 := hrs.reader(); err2 == nil {
 			return n, nil
 		}
-	} else if err == io.EOF {
+	case io.EOF:
```

Comment:
Non-blocking (as it's an existing issue), but we should consider adding a `default` to these switches to be explicit about conditions not included in the list; that would also allow the `exhaustive` linter to be enabled, which can catch cases where new options were forgotten; https://golangci-lint.run/usage/linters/#exhaustive

We should also consider enabling the `nakedret` linter; naked returns are err-prone (more so in this case where `err` is used as a named return)